compiler/rustc_passes/src/dead.rs

```diff
@@ -508,48 +485,57 @@ impl<'tcx> MarkSymbolVisitor<'tcx> {
         }
     }
 
-    fn solve_rest_impl_items(&mut self, mut unsolved_impl_items: Vec<(hir::ItemId, LocalDefId)>) {
-        let mut ready;
-        (ready, unsolved_impl_items) =
-            unsolved_impl_items.into_iter().partition(|&(impl_id, impl_item_id)| {
-                self.impl_item_with_used_self(impl_id, impl_item_id)
-            });
-
-        while !ready.is_empty() {
-            self.worklist =
-                ready.into_iter().map(|(_, id)| (id, ComesFromAllowExpect::No)).collect();
-            self.mark_live_symbols();
-
-            (ready, unsolved_impl_items) =
-                unsolved_impl_items.into_iter().partition(|&(impl_id, impl_item_id)| {
-                    self.impl_item_with_used_self(impl_id, impl_item_id)
-                });
+    /// Returns an impl or impl item should be checked or not
```

Comment:
Checked for what?
I assumed that the "checked == false" items will never report the dead code lint, and "checked == true" items *may* report the dead code lint later.

However, this code below
```rust
                && !self.live_symbols.contains(&trait_def_id)
            {
                return false;
```
says that some dead items should never be reported, which contradicts the intuition above.

Reply:
Updated the method to `check_impl_or_impl_item_live` and the comment to `Returns whether `local_def_id` is live or not.`

The logic here is just to check the given local_def_id is live or not.